server/src/main/java/io/crate/rest/action/SqlHttpHandler.java

```diff
@@ -119,11 +120,13 @@ protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request)
                 request);
             Map<String, List<String>> parameters = new QueryStringDecoder(request.uri()).parameters();
             ByteBuf content = request.content();
-            handleSQLRequest(session, content, paramContainFlag(parameters, "types"))
-                .whenComplete((result, t) -> {
+            ByteBuf resultBuffer = ctx.alloc().buffer();
+            handleSQLRequest(session, resultBuffer, content, paramContainFlag(parameters, "types"))
+                .whenComplete((_, t) -> {
                     try {
-                        sendResponse(session, ctx, request, parameters, result, t);
+                        sendResponse(session, ctx, request, parameters, resultBuffer, t);
                     } catch (Throwable ex) {
+                        resultBuffer.release();
                         LOGGER.error("Error sending response", ex);
```

Comment:
Wouldn't it be safer to release only in `finally`?

Afaik we can end up with double release: first when we release in `sendResponse` and then when any line below in `sendResponse` throws, we catch it here and release again

Reply:
> it always get released twice

I meant releasing only in finally and remove `release` in `sendResponse`.

>The catch case is meant to run when the write didn't happen.

I'm talking about the scenario when write didn't happen but hits non-happy code path
```
sendResponse() {
      result.release(); // --->  released once
      var sessionSettings = session.sessionSettings();
      AccessControl accessControl = ...
      ...
        } catch (IOException e) {
              // For example this throws
              throw new RuntimeException(e);  // -> this is caught and we again do release() in the caller
       }
}
```

Reply:
> I meant releasing only in finally and remove release in sendResponse.

We can't do that because writing out `DefaultFullHttpResponse` will release the underyling content buffer.

But I see now what you mean with the double release case. Will create a PR with a fix